src/components/scrollbar.ts

```diff
@@ -0,0 +1,565 @@
+    }
+
+    this.scrollTrackLength = this.scrollTrackLength === 0
+      ? this.scrollable.containerBounds[this.dimension]
+      : this.scrollTrackLength;
+    this.contentInitialPosition = this.scrollable.content.position[this.axis] as number;
+    this.containerSize = this.scrollable.containerBounds[this.dimension];
+
+    this.track = this.createTrack(this.scrollTrackLength, DEFAULT_SCROLLBAR_THICKNESS);
+    this.scrollbar = this.createScrollbar(this.getProportionalLength(), DEFAULT_SCROLLBAR_THICKNESS);
+    // extend entire scrollbar hit area based on hit tolerance to make interaction easier. used for onClick event
+    const extendedHitArea = this.extendHitArea();
+
+    this.onMouseEnter = this.mouseEnter;
+    this.onMouseLeave = this.mouseLeave;
+    this.onMouseMove = this.scrollbarHitTest;
+    this.onMouseDown = this.scrollbarHitTest;
+    this.onClick = this.trackClick;
+    this.scrollbarDrag = this.scrollbarDragTool();
+    this.arrowKeyDown = this.arrowKeyDownTool();
+  }
+
+  /**
+   * Activate the default tool. Used when the scrollable container is hovered or active.
+   */
+  activateDefaultTool(): void {
+    this.arrowKeyDown.activate();
+  }
+
+  /**
+   * Handler for the wheel event.
+   * PaperJS does not have a scroll event handler, so this is set up externally where the HTML canvas element can be
+   * accessed.
+   * @param event WheelEvent passed from the HTML canvas.
+   *
+   * @example
+   * const scrollbar = new ScrollbarComponent(...);
+   * const canvas = this.demo.canvas.nativeElement;
+   * canvas.onwheel = (event: WheelEvent) => {
+   *   scrollbar.onScroll(event);
+   * };
+   */
+  onScroll(event: WheelEvent): void {
+    event.preventDefault();
+    const validScrollDirection = this.isHorizontal ? event.deltaX !== 0 : event.deltaY !== 0;
+    if (!validScrollDirection) {
+      return;
+    }
+    event.preventDefault();
+    this.setActiveContinuously();
+    this.isHorizontal
+      ? this.changeScrollAndContentPosition(this.scrollbar.position.x + event.deltaX)
+      : this.changeScrollAndContentPosition(this.scrollbar.position.y + event.deltaY);
+  }
+
+  /**
+   * Disable the default effects.
+   */
+  disableDefaultEffects(): void {
+    this.enableDefaultEffects = false;
+  }
+
+  /**
+   * Set custom active and normal effects from the parent.
+   * Use with disableDefaultEffects for fuller control. Effects can target ScrollbarComponent elements or something
+   * external.
+   * @param customEffects
+   *
+   * @example
+   * const scrollbar = newScrollbarComponent(...);
+   * scrollbar.setCustomEffects({
+   *   setActive: () => {
+   *     scrollbar.getScrollbar().fillColor = 'green';
+   *   },
+   *   setNormal: () => {
+   *     scrollbar.getScrollbar().fillColor = 'red;
+   *   }
+   * });
+   */
+  setCustomEffects(customEffects: CustomEffects): void {
+    this.customEffects = customEffects;
+  }
+
+  /**
+   * Gets the scrollbar element.
+   */
+  getScrollbar(): paper.Path {
```

Comment:
It's better to be explicit about the possibility that a return value may be undefined. e.g. you could express the return type as a type union using `paper.Path | undefined`. It is also nice to document the conditions under which an undefined value is expected in the tsdoc.

Reply:
fixed. should always return a scrollbar now.